tools/src/main/java/org/apache/kafka/tools/ReplicaVerificationTool.java

```diff
@@ -0,0 +1,740 @@
+                return consumer.endOffsets(topicPartitions).entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            } else if (ListOffsetsRequest.EARLIEST_TIMESTAMP == initialOffsetTime) {
+                return consumer.beginningOffsets(topicPartitions).entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            } else {
+                Map<TopicPartition, Long> timestampsToSearch = topicPartitions.stream()
+                    .collect(Collectors.toMap(Function.identity(), tp -> initialOffsetTime));
+                return consumer.offsetsForTimes(timestampsToSearch).entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().offset()));
+            }
+        }
+    }
+
+    private static Properties consumerConfig(String brokerUrl) {
+        Properties properties = new Properties();
+        properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerUrl);
+        properties.put(ConsumerConfig.GROUP_ID_CONFIG, "ReplicaVerification");
+        properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        return properties;
+    }
+
+    private static Map<Integer, Node> brokerDetails(Admin adminClient) throws ExecutionException, InterruptedException {
+        return adminClient.describeCluster().nodes().get().stream().collect(Collectors.toMap(Node::id, Function.identity()));
+    }
+
+    private static Collection<TopicDescription> listTopicsMetadata(Admin adminClient) throws ExecutionException, InterruptedException {
+        Set<String> topics = adminClient.listTopics(new ListTopicsOptions().listInternal(true)).names().get();
+        return adminClient.describeTopics(topics).allTopicNames().get().values();
+    }
+
+    private static Admin createAdminClient(String bootstrapServer) {
+        Properties props = new Properties();
+        props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
+        return Admin.create(props);
+    }
+
+    private static class ReplicaVerificationToolOptions extends CommandDefaultOptions {
+        private final OptionSpec<String> brokerListOpt;
+        private final OptionSpec<String> bootstrapServerOpt;
+        private final OptionSpec<Integer> fetchSizeOpt;
+        private final OptionSpec<Integer> maxWaitMsOpt;
+        private final OptionSpec<String> topicWhiteListOpt;
+        private final OptionSpec<String> topicsIncludeOpt;
+        private final OptionSpec<Long> initialOffsetTimeOpt;
+        private final OptionSpec<Long> reportIntervalOpt;
+
+        ReplicaVerificationToolOptions(String[] args) {
+            super(args);
+            bootstrapServerOpt = parser.accepts("bootstrap-server",
```

Comment:
Looks like there is one already: [KIP-752](https://cwiki.apache.org/confluence/display/KAFKA/KIP-752%3A+Support+--bootstrap-server+in+ReplicaVerificationTool). It didn't have much attention, so I would rather continue with this improvement, or remove it. We can also try to revamp that thread, but this is something that the community agreed on long time ago (KIP-999), and this tool was just an oversight.

Reply:
Does the tool actually work? Looking at the discussion thread for KIP-752 it seems it may not work anymore. If it does not work, we should just deprecate it and not bother migrating it.

Reply:
Removing extra changes.